manifest/store.go

```diff
@@ -0,0 +1,115 @@
+package manifest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/manifest"
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/opencontainers/go-digest"
+)
+
+// Store provides methods which can operate on a manifest store.
+type Store interface {
+	Get(digest.Digest) (*distribution.Manifest, error)
+	Set(manifest distribution.Manifest) (digest.Digest, error)
```

Comment:
I wouldn't recommend on having the manifest store do the actual serialization. That binds this interface to supporting round trips. This should probably just store the actual bytes received.